examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -1411,34 +1168,34 @@ def compute_time_ids(original_size, crops_coords_top_left):
 
                 # Gather the losses across all processes for logging (if we use distributed training).
                 avg_loss = accelerator.gather(loss.repeat(args.train_batch_size)).mean()
-                train_loss += avg_loss.item() / args.gradient_accumulation_steps
-
+                train_loss += avg_loss / args.gradient_accumulation_steps
                 # Backpropagate
-                # TODO: check why this cause bufferoverflow issue
-                # with accelerator.autocast():
+                #TODO: check why this cause bufferoverflow issue
+                #with torch.autocast(device_type="hpu", dtype=weight_dtype, enabled=True):
                 accelerator.backward(loss)
                 htcore.mark_step()
 
                 if accelerator.sync_gradients:
-                    if args.finetuning_method == "lora":
-                        accelerator.clip_grad_norm_(params_to_optimize, args.max_grad_norm)
+                    params_to_clip = unet.parameters()
+                    if gaudi_config.use_fused_clip_norm:
+                        fused_clip_norm.clip_norm(params_to_clip)
                     else:
-                        params_to_clip = unet.parameters()
                         accelerator.clip_grad_norm_(params_to_clip, args.max_grad_norm)
+
                 optimizer.step()
                 lr_scheduler.step()
                 optimizer.zero_grad(set_to_none=True)
                 htcore.mark_step()
+                hb_profiler.step()
+
 
             # Checks if the accelerator has performed an optimization step behind the scenes
             if accelerator.sync_gradients:
                 progress_bar.update(1)
                 global_step += 1
-                accelerator.log({"train_loss": train_loss}, step=global_step)
-                train_loss = 0.0
 
                 if accelerator.is_main_process:
-                    if args.checkpointing_steps is not None and global_step % args.checkpointing_steps == 0:
+                    if global_step % args.checkpointing_steps == 0:
```

Comment:
should we change to left as
if args.checkpointing_steps is not None and global_step % args.checkpointing_steps == 0:

Reply:
From the original change, checkpointing_steps default set to None.
But ours is 500. Do you want to change default value and add the same condition check here?